src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/SessionAndEntryService.kt

```diff
@@ -0,0 +1,89 @@
+package uk.gov.justice.digital.hmpps.keyworker.services
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerEntry
+import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerEntryRepository
+import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerInteraction
+import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerSession
+import uk.gov.justice.digital.hmpps.keyworker.domain.KeyworkerSessionRepository
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.CaseNote
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.CaseNote.Companion.ENTRY_SUBTYPE
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.CaseNote.Companion.SESSION_SUBTYPE
+import uk.gov.justice.digital.hmpps.keyworker.integration.casenotes.CaseNotesApiClient
+import uk.gov.justice.digital.hmpps.keyworker.integration.events.CaseNoteInformation
+import java.util.UUID
+
+@Transactional
+@Service
+class SessionAndEntryService(
+  private val caseNoteApi: CaseNotesApiClient,
+  private val ksRepository: KeyworkerSessionRepository,
+  private val keRepository: KeyworkerEntryRepository,
+) {
+  fun new(personInfo: PersonInformation) {
+    saveKeyworkerInteraction(getCaseNote(personInfo).asKeyworkerInteraction())
+  }
+
+  fun update(personInfo: PersonInformation) {
+    updateKeyworkerInteraction(getCaseNote(personInfo)) {
+      apply {
+        occurredAt = it.occurredAt
+      }
+    }
+  }
+
+  fun move(personInfo: PersonInformation) {
+    updateKeyworkerInteraction(getCaseNote(personInfo)) {
+      apply {
+        personIdentifier = it.personIdentifier
+      }
+    }
+  }
+
+  fun delete(personInfo: PersonInformation) {
+    when (personInfo.info.subType) {
+      SESSION_SUBTYPE -> ksRepository.deleteById(personInfo.info.id)
```

Comment:
Wondering about the value of having separate tables for sessions and entries vs a type of some sort. UR has shown that many users want to remove the KW entry concept completely. Can't see that happening but it does suggest that the direction of travel is fewer KW event types rather than possibly adding a new one

Reply:
Well argued and I agree